### DISCUSSION TITLE
New window are not in focus

### DISCUSSION THREAD

**Discussion by hamadahiro** [asmvik/yabai#2481] [Q&A]
If I open a new window that is set to be open in a certain space, and I'm in another space, it will open the new window unfocused.

For example if the app is opening in space 2, and I'm opening it while in space 2, it will focus to space 2, but the window will not be in focus. 

Here's my config:

```
# load scripting additions
yabai -m signal --add event=dock_did_restart action="sudo yabai --load-sa"
yabai -m signal --add event=window_focused action="sketchybar --trigger window_focus"
yabai -m signal --add event=window_destroyed active=yes action="yabai -m query --windows --window &> /dev/null || yabai -m window --focus mouse &> /dev/null || yabai -m window --focus \$(yabai -m query --windows --space | jq .[0].id) &> /dev/null"
sudo yabai --load-sa

# disable macos window manager
launchctl unload -F /System/Library/LaunchAgents/com.apple.WindowManager.plist > /dev/null 2>&1 &

# kills animation when opening a folder on a desktop
defaults write com.apple.finder DisableAllAnimations -bool true
killall Finder # or logout and login
```

**Comment by hamadahiro** [comment]
I think I found the solution:

`yabai -m signal --add event=window_created action="yabai -m query --windows --window &> /dev/null || yabai -m window --focus mouse &> /dev/null || yabai -m space --focus \$(yabai -m query --windows --space | jq .[0].space)"`

**Comment by Bellavene** [comment]
I don't use spaces much, but have a similar problem where hotkey windows of some apps as they get visible are half focused for some reason, which breaks some things with current focus.
I've managed it like that
`yabai -m signal --add event=window_created app='iTerm' action="osascript -e 'tell application \"iTerm\" to activate'"`
